api_fhir/models/contract.py

```diff
@@ -0,0 +1,5 @@
+from api_fhir.models import DomainResource
+
+
+class Contract(DomainResource):
+    pass
```

Comment:
I believe there you need to add the Resource fields like in the other resources.
https://www.hl7.org/fhir/stu3/contract.html

Reply:
Ideally well... but from the bound stored proc, I don't have the Contract details... so at the end, all I provide is a contract 'reference':
`{
    "resourceType": "EligibilityResponse",
    "insurance": [
        {
            "benefitBalance": [
                {
                    "category": "medical",
                    "financial": [
                        {
                            "allowedMoney": {
                                "value": 0
                            },
                            "usedMoney": {
                                "value": 0
                            }
                        }
                    ]
                }
            ],
            "contract": {
                "reference": "Contract/BCUL0001/2019-12-31 00:00:00"
            }
        }
    ]
}`

Reply:
First of all I believe the Contract reference ID should look rather "BCUL0001" that instead "BCUL0001/2019-12-31 00:00:00". You can change this by extending the ReferenceConverterMixin like here:
https://github.com/openimis/openimis-be-api_fhir_py/blob/master/api_fhir/converters/patientConverter.py#L12

Secondly I am not sure but when we have that kind of reference probably we should expose that resource because the client will expect that. So probably we need to create FHIR mapping for that.

Reply:
I would also have preferred to have a 'straight' unique ID for the contract..., but BCUL0001 is in fact the product code, so it is not enough to identify a policy.

imis> select pol.PolicyId, pol.ProdId, prod.ProductCode, pol.expiryDate from tblPolicy pol, tblProduct prod  where pol.familyid = 1 and pol.ProdId = prod.ProdId;
+------------+----------+---------------+--------------+
| PolicyId   | ProdId   | ProductCode   | expiryDate   |
|------------+----------+---------------+--------------|
| 1          | 4        | BCUL0001      | 2018-12-19   |
| 2          | 4        | BCUL0001      | 2019-12-19   |
+------------+----------+---------------+--------------+

... I believe one 'natural identifier' for a policy is the combination of the product code and the policy expiry date... hence the proposal...

I do agree for the second point: we should be exposing the Contract as 'complete' FHIR resource, but I don't have the budget to do it now... (and it is not necessary for the odoo/bahmni integration right now...)
>> will have to be done in another integration scope...

Reply:
We should be careful with quick fix only for one project needs. If we really need it for Bahmni integration, I would suggest to make a release without this change and next release, e.g. 1.0.1 with adding not completed Contract. 

Adding not completed Contract resource in 1.0 release is not a good idea in my opinion.

Reply:
decision from our weekly meeting is to release with the reference... and complete the contract FHIR resource in the scope of the project that requires it...